### PR TITLE
TINY-11551: Upgrade Typescript to v5.7

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: 

Description of Changes:
* Upgraded Typescript to v5.7.2
* Commented out ineffective code in `ControlSelection.ts`
* Added `layerX` and `layerY` properties to drag event but defaulted them to 0 as [they are not standard properties](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/layerX)  

Pre-checks:
* [x] ~Changelog entry added~
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

## Summary by CodeRabbit

- **New Features**
	- Enhanced type safety in configuration handling for behavior creation.
	- Improved drag-and-drop event handling with additional mouse position properties.

- **Bug Fixes**
	- Adjusted ghost element positioning logic during resizing to address potential issues.

- **Tests**
	- Updated assertions in caret movement tests to improve type handling.

- **Chores**
	- Updated TypeScript version in development dependencies.

### CODE REVIEW

File: modules/tinymce/src/core/main/ts/api/dom/ControlSelection.ts [hunk 1]
```diff
@@ -223,15 +223,17 @@ const ControlSelection = (selection: EditorSelection, editor: Editor): ControlSe
...
 
-    // Update ghost Y position if needed
-    if (selectedHandle[3] < 0 && selectedElmGhost.clientHeight <= height) {
-      dom.setStyle(selectedElmGhost, 'top', selectedElmY + (startH - height));
-    }
+      // Update ghost Y position if needed
+      if (selectedHandle[3] < 0 && selectedElmGhost.clientHeight <= height) {
+        dom.setStyle(selectedElmGhost, 'top', selectedElmY + (startH - height));
+      }
+    */
```
[the remark this answers is not in the document]
Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/api/dom/ControlSelection.ts:236]: @coderabbitai ignore. Will be addressed in a future ticket, TINY-11702

File: modules/tinymce/src/core/test/ts/browser/caret/CaretUtilsTest.ts [hunk 2]
```diff
@@ -42,18 +42,18 @@ describe('browser.tinymce.core.CaretUtilTest', () => {
 
   it('isForwards', () => {
     assert.isTrue(CaretUtils.isForwards(1));
-    assert.isTrue(CaretUtils.isForwards(10));
-    assert.isFalse(CaretUtils.isForwards(0));
+    assert.isTrue(CaretUtils.isForwards(10 as any));
+    assert.isFalse(CaretUtils.isForwards(0 as any));
     assert.isFalse(CaretUtils.isForwards(-1));
-    assert.isFalse(CaretUtils.isForwards(-10));
+    assert.isFalse(CaretUtils.isForwards(-10 as any));
```
[the remark this answers is not in the document]
Comment (TheSpyder) [reply modules/tinymce/src/core/test/ts/browser/caret/CaretUtilsTest.ts:48]: @coderabbitai the `HDirection` enum only accepts 1 and -1. The values are invalid.

@tiny-ben-tran just delete the invalid cases; they are from a land before types.

File: yarn.lock [hunk 3]
```diff
@@ -11536,11 +11536,16 @@ typedarray@^0.0.6:
   resolved "https://registry.yarnpkg.com/typescript/-/typescript-5.3.3.tgz#b3ce6ba258e72e6305ba66f5c9b452aaee3ffe37"
   integrity sha512-pXWcraxM0uxAS+tN0AG/BF2TyqmHO014Z070UsJ+pFvYuRSq8KH8DmWpnbXe0pEPDHXZV3FcAbJkijJ5oNEnWw==
 
-typescript@^4.8.3, typescript@^4.9.5:
+typescript@^4.9.5:
```
Comment (TheSpyder) [inline yarn.lock:11539]: derp we should probably upgrade `eslint-plugin` as well, that seems to be where this comes from (use `yarn why typescript` to confirm)

**Review by TheSpyder — APPROVED** [review]

**Review by spocke — APPROVED** [review]

File: yarn.lock (hunk 3 above)
Comment (tiny-ben-tran) [reply yarn.lock:11539]: @ltrouton is upgrading Typescript in eslint-plugin.

**Comment by tiny-ben-tran** [comment]
@TheSpyder The two properties were added in [this PR](https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1628) but there is no explanation to why. Also the two properties are non-standard https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/layerX

**Pull request merged**